In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import os
os.chdir('/content/drive/MyDrive/A2W/디도 과제')

In [52]:
import pandas as pd
# 소수점 2자리까지 표시
pd.options.display.float_format = '{:.2f}'.format

df = pd.read_csv('soil_land.csv')
df

umd  sgg      umd2  sum_cnt  ...  SLDPT_TPCD  SCSTX_CD  SLTP_CD  EIGHT_AGL_dir
0     개진면  고령군   고령군 개진면     0.00  ...          20         2        1              S
1     성산면  고령군   고령군 성산면     0.00  ...          10         2        1              W
2     우곡면  고령군   고령군 우곡면     1.00  ...          10         3       13              W
3     각북면  청도군   청도군 각북면     0.00  ...          10         2        1              N
4     이서면  청도군   청도군 이서면     0.00  ...          10         2        1              E
..    ...  ...       ...      ...  ...         ...       ...      ...            ...
632   신안동  진주시   진주시 신안동     0.00  ...           0         0        0              N
633  충무공동  진주시  진주시 충무공동     0.00  ...           0         0        0              N
634   욕지면  통영시   통영시 욕지면     0.00  ...           0         0        0              N
635   사량면  통영시   통영시 사량면     0.00  ...           0         0        0              N
636   양주동  양산시   양산시 양주동     0.00  ...           0         0        0              N

[637 rows x 17 columns]

In [63]:
# 범주형 변수 분석 함수
def check_categorical(check_data, check_data_vc):
  feature = check_data_vc.columns[0]
  col1 =  feature, 'land0'
  col2 =  feature, 'land1'

  df1 = pd.DataFrame(check_data.columns[:-1], columns = [feature])


  values = check_data_vc[check_data_vc['landslide']==1].value_counts().tolist()
  keys = check_data_vc[check_data_vc['landslide']==1].value_counts().index.tolist()
  indexs = [keys_index[0] for keys_index in keys]

  A1 = dict(zip(indexs, values)) # 1

  df2 = pd.DataFrame(list(A1.items()), columns = col2)


  values2 = check_data_vc[check_data_vc['landslide']==0].value_counts().tolist()
  keys2 = check_data_vc[check_data_vc['landslide']==0].value_counts().index.tolist()
  indexs2 = [keys_index[0] for keys_index in keys2]

  A2 = dict(zip(indexs2, values2)) # 1

  df3 = pd.DataFrame(list(A2.items()), columns = col1)

  df_merge = pd.merge(df1,df2, how='outer',on=feature)
  df_merge = pd.merge(df_merge,df3, how='outer',on=feature)
  df_merge = df_merge.fillna(0)

  df_merge['ratio'] = df_merge['land1'] / (df_merge['land1'] + df_merge['land0'])

  display(df_merge)

  return df_merge

In [81]:
import matplotlib.pyplot as plt

def plot(df_merge):
  fig = plt.figure(figsize=(15,4))
  ax1 = fig.add_subplot(1, 5, 1)
  ax1.plot(, y)
  ax2 = fig.add_subplot(1, 5, 3)
  ax3 = fig.add_subplot(1, 5, 5)
  plt.show()

In [ ]:
fig = plt.figure()
In [53]: ax1 = fig.add_subplot(1, 2, 1)
In [54]: ax2 = fig.add_subplot(1, 2, 2)
In [55]: plt.show()

# 1. 'PRRCK_LARG' - 범주형

각각의 범주형 데이터 상관계수 확인하기



In [55]:
land = pd.DataFrame(df['landslide'])
land

landslide
0         0.00
1         0.00
2         1.00
3         0.00
4         0.00
..         ...
632       0.00
633       0.00
634       0.00
635       0.00
636       0.00

[637 rows x 1 columns]

In [56]:
# 원-핫 인코딩으로 만들어 줌
dumm1 = pd.get_dummies(df['PRRCK_LARG'])
dumm1

0  1  2  3
0    0  0  1  0
1    0  0  1  0
2    0  0  1  0
3    0  1  0  0
4    0  0  1  0
..  .. .. .. ..
632  1  0  0  0
633  1  0  0  0
634  1  0  0  0
635  1  0  0  0
636  1  0  0  0

[637 rows x 4 columns]

In [57]:
# 데이터를 붙여줌
check_data1 = dumm1.join(land, how='inner')
check_data1

0  1  2  3  landslide
0    0  0  1  0       0.00
1    0  0  1  0       0.00
2    0  0  1  0       1.00
3    0  1  0  0       0.00
4    0  0  1  0       0.00
..  .. .. .. ..        ...
632  1  0  0  0       0.00
633  1  0  0  0       0.00
634  1  0  0  0       0.00
635  1  0  0  0       0.00
636  1  0  0  0       0.00

[637 rows x 5 columns]

In [58]:
check_data1.corr()

0     1     2     3  landslide
0          1.00 -0.16 -0.17 -0.09      -0.07
1         -0.16  1.00 -0.65 -0.36       0.01
2         -0.17 -0.65  1.00 -0.38      -0.05
3         -0.09 -0.36 -0.38  1.00       0.08
landslide -0.07  0.01 -0.05  0.08       1.00

In [59]:
check_data1_vc = df[['PRRCK_LARG', 'landslide']]

In [60]:
check_data1_vc[check_data1_vc['landslide']==1].value_counts()

PRRCK_LARG  landslide
1           1.00         92
2           1.00         91
3           1.00         52
0           1.00          5
dtype: int64

In [61]:
check_data1_vc[check_data1_vc['landslide']==0].value_counts()

PRRCK_LARG  landslide
2           0.00         170
1           0.00         147
3           0.00          60
0           0.00          20
dtype: int64

In [64]:
check_categorical(check_data1, check_data1_vc)

PRRCK_LARG  land1  land0  ratio
0           0      5     20   0.20
1           1     92    147   0.38
2           2     91    170   0.35
3           3     52     60   0.46

PRRCK_LARGdms 0.1을 넘는 변수가 없다. 산사태가 발생한 1, 2에 동등하게 많이 분포되어 있기 때문에 연관성을 찾을 수 없다.

전체적으로 몇개 데이터 중 몇개가 데이터 비중이 있는지
1, 2, 3, 0 : 91/(170+91) = %

# 2. 'PRRCK_MDDL' - 범주형

In [65]:
# 원-핫 인코딩으로 만들어 줌
dumm2 = pd.get_dummies(df['PRRCK_MDDL'])
dumm2

0   11  12  14  15  16  21  22  23  24  25  26  31  32
0     0   0   0   0   0   0   0   0   0   1   0   0   0   0
1     0   0   0   0   0   0   0   0   0   0   0   1   0   0
2     0   0   0   0   0   0   0   0   0   1   0   0   0   0
3     0   0   1   0   0   0   0   0   0   0   0   0   0   0
4     0   0   0   0   0   0   0   0   0   0   0   1   0   0
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..
632   1   0   0   0   0   0   0   0   0   0   0   0   0   0
633   1   0   0   0   0   0   0   0   0   0   0   0   0   0
634   1   0   0   0   0   0   0   0   0   0   0   0   0   0
635   1   0   0   0   0   0   0   0   0   0   0   0   0   0
636   1   0   0   0   0   0   0   0   0   0   0   0   0   0

[637 rows x 14 columns]

In [66]:
# 데이터를 붙여줌
check_data2 = dumm2.join(land, how='inner')
check_data2

0  11  12  14  15  16  21  22  23  24  25  26  31  32  landslide
0    0   0   0   0   0   0   0   0   0   1   0   0   0   0       0.00
1    0   0   0   0   0   0   0   0   0   0   0   1   0   0       0.00
2    0   0   0   0   0   0   0   0   0   1   0   0   0   0       1.00
3    0   0   1   0   0   0   0   0   0   0   0   0   0   0       0.00
4    0   0   0   0   0   0   0   0   0   0   0   1   0   0       0.00
..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..        ...
632  1   0   0   0   0   0   0   0   0   0   0   0   0   0       0.00
633  1   0   0   0   0   0   0   0   0   0   0   0   0   0       0.00
634  1   0   0   0   0   0   0   0   0   0   0   0   0   0       0.00
635  1   0   0   0   0   0   0   0   0   0   0   0   0   0       0.00
636  1   0   0   0   0   0   0   0   0   0   0   0   0   0       0.00

[637 rows x 15 columns]

In [67]:
check_data2.corr()

0    11    12    14    15  ...    25    26    31    32  landslide
0          1.00 -0.11 -0.03 -0.07 -0.01  ... -0.05 -0.04 -0.09 -0.02      -0.07
11        -0.11  1.00 -0.09 -0.18 -0.02  ... -0.12 -0.10 -0.23 -0.06      -0.01
12        -0.03 -0.09  1.00 -0.05 -0.01  ... -0.04 -0.03 -0.07 -0.02       0.03
14        -0.07 -0.18 -0.05  1.00 -0.01  ... -0.08 -0.06 -0.15 -0.04      -0.01
15        -0.01 -0.02 -0.01 -0.01  1.00  ... -0.01 -0.01 -0.02 -0.00      -0.03
16        -0.04 -0.09 -0.03 -0.06 -0.01  ... -0.04 -0.03 -0.08 -0.02       0.05
21        -0.01 -0.04 -0.01 -0.02 -0.00  ... -0.02 -0.01 -0.03 -0.01      -0.05
22        -0.08 -0.21 -0.07 -0.13 -0.02  ... -0.09 -0.08 -0.17 -0.05       0.03
23        -0.04 -0.11 -0.03 -0.07 -0.01  ... -0.05 -0.04 -0.09 -0.02       0.01
24        -0.08 -0.22 -0.07 -0.14 -0.02  ... -0.09 -0.08 -0.18 -0.05      -0.14
25        -0.05 -0.12 -0.04 -0.08 -0.01  ...  1.00 -0.04 -0.10 -0.03       0.01
26        -0.04 -0.10 -0.03 -0.06 -0.01  ... -0.04  1.00 -0.09 -0.02       0.06
31        -0.09 -0.23 -0.07 -0.15 -0.02  ... -0.10 -0.09  1.00 -0.05       0.12
32        -0.02 -0.06 -0.02 -0.04 -0.00  ... -0.03 -0.02 -0.05  1.00      -0.09
landslide -0.07 -0.01  0.03 -0.01 -0.03  ...  0.01  0.06  0.12 -0.09       1.00

[15 rows x 15 columns]

In [68]:
check_data2_vc = df[['PRRCK_MDDL', 'landslide']]

In [69]:
check_data2_vc[check_data2_vc['landslide']==1].value_counts()

PRRCK_MDDL  landslide
31          1.00         52
11          1.00         51
22          1.00         36
14          1.00         23
24          1.00         20
25          1.00         13
26          1.00         12
23          1.00         10
16          1.00         10
12          1.00          8
0           1.00          5
dtype: int64

In [70]:
check_data2_vc[check_data2_vc['landslide']==0].value_counts()

PRRCK_MDDL  landslide
11          0.00         88
24          0.00         72
31          0.00         51
22          0.00         50
14          0.00         40
0           0.00         20
25          0.00         19
23          0.00         15
26          0.00         11
32          0.00          9
16          0.00          9
12          0.00          9
21          0.00          3
15          0.00          1
dtype: int64

In [71]:
check_categorical(check_data2, check_data2_vc)

PRRCK_MDDL  land1  land0  ratio
0            0   5.00     20   0.20
1           11  51.00     88   0.37
2           12   8.00      9   0.47
3           14  23.00     40   0.37
4           15   0.00      1   0.00
5           16  10.00      9   0.53
6           21   0.00      3   0.00
7           22  36.00     50   0.42
8           23  10.00     15   0.40
9           24  20.00     72   0.22
10          25  13.00     19   0.41
11          26  12.00     11   0.52
12          31  52.00     51   0.50
13          32   0.00      9   0.00

PRRCK_LARG는 31과 11이 발생했을 때 산사태가 많이 발생했다. 31은 양의 상관관계, 11은 음의 상관관계를 갖게 된다.

# 3. 'LOCTN_ALTT' - 수치형

In [ ]:
# 원-핫 인코딩으로 만들어 줌
dumm3 = df[['LOCTN_ALTT', 'landslide']]# 데이터를 붙여줌
dumm3

LOCTN_ALTT  landslide
0         306.1        0.0
1         103.2        0.0
2          64.1        1.0
3         573.4        0.0
4         249.5        0.0
..          ...        ...
632         0.0        0.0
633         0.0        0.0
634         0.0        0.0
635         0.0        0.0
636         0.0        0.0

[637 rows x 2 columns]

고도는 numerical 값이다. 따라서 상관계수 분석을 산사태와 비교할 필요 없다.

In [ ]:
dumm3.corr()

LOCTN_ALTT  landslide
LOCTN_ALTT    1.000000   0.176618
landslide     0.176618   1.000000

위와 같이 0.17이라는 상관관계가 나왔다.

# 4. 'LOCTN_GRDN' - 수치형

In [ ]:
# 원-핫 인코딩으로 만들어 줌
dumm4 = pd.get_dummies(df['LOCTN_GRDN'])
dumm4

0.0   0.5   0.7   0.9   1.1   1.2   ...  43.1  45.8  47.1  49.4  50.4  51.9
0       0     0     0     0     0     0  ...     0     0     0     0     0     0
1       0     0     0     0     0     0  ...     0     0     0     0     0     0
2       0     0     0     0     0     0  ...     0     0     0     0     0     0
3       0     0     0     0     0     0  ...     0     0     0     0     0     0
4       0     0     0     0     0     0  ...     0     0     0     0     0     0
..    ...   ...   ...   ...   ...   ...  ...   ...   ...   ...   ...   ...   ...
632     1     0     0     0     0     0  ...     0     0     0     0     0     0
633     1     0     0     0     0     0  ...     0     0     0     0     0     0
634     1     0     0     0     0     0  ...     0     0     0     0     0     0
635     1     0     0     0     0     0  ...     0     0     0     0     0     0
636     1     0     0     0     0     0  ...     0     0     0     0     0     0

[637 rows x 264 columns]

In [ ]:
# 데이터를 붙여줌
dumm4 = df[['LOCTN_GRDN', 'landslide']]# 데이터를 붙여줌
dumm4

LOCTN_GRDN  landslide
0          18.7        0.0
1          13.4        0.0
2          17.7        1.0
3          22.5        0.0
4          15.1        0.0
..          ...        ...
632         0.0        0.0
633         0.0        0.0
634         0.0        0.0
635         0.0        0.0
636         0.0        0.0

[637 rows x 2 columns]

In [ ]:
dumm4.corr()

LOCTN_GRDN  landslide
LOCTN_GRDN    1.000000   0.156662
landslide     0.156662   1.000000

0.15라는 상관관계 도출

# 5. 'CLZN_CD' - 범주형

In [73]:
# 원-핫 인코딩으로 만들어 줌
dumm5 = pd.get_dummies(df['CLZN_CD'])
dumm5

0  1  2  3  4
0    0  0  0  1  0
1    0  0  0  1  0
2    0  0  0  1  0
3    0  0  0  1  0
4    0  0  0  1  0
..  .. .. .. .. ..
632  1  0  0  0  0
633  1  0  0  0  0
634  1  0  0  0  0
635  1  0  0  0  0
636  1  0  0  0  0

[637 rows x 5 columns]

In [74]:
# 데이터를 붙여줌
check_data5 = dumm5.join(land, how='inner')
check_data5

0  1  2  3  4  landslide
0    0  0  0  1  0       0.00
1    0  0  0  1  0       0.00
2    0  0  0  1  0       1.00
3    0  0  0  1  0       0.00
4    0  0  0  1  0       0.00
..  .. .. .. .. ..        ...
632  1  0  0  0  0       0.00
633  1  0  0  0  0       0.00
634  1  0  0  0  0       0.00
635  1  0  0  0  0       0.00
636  1  0  0  0  0       0.00

[637 rows x 6 columns]

In [75]:
check_data5.corr()

0     1     2     3     4  landslide
0          1.00 -0.03 -0.15 -0.12 -0.14      -0.07
1         -0.03  1.00 -0.13 -0.10 -0.12      -0.02
2         -0.15 -0.13  1.00 -0.44 -0.52      -0.14
3         -0.12 -0.10 -0.44  1.00 -0.39       0.15
4         -0.14 -0.12 -0.52 -0.39  1.00       0.05
landslide -0.07 -0.02 -0.14  0.15  0.05       1.00

In [76]:
check_data5_vc = df[['CLZN_CD', 'landslide']]

In [79]:
check_categorical(check_data5, check_data5_vc)

CLZN_CD  land1  land0  ratio
0        0      5     20   0.20
1        1      6     12   0.33
2        2     67    167   0.29
3        3     80     80   0.50
4        4     82    118   0.41

좀 묶기가 애매해보인다..

# 6. 'TPGRP_TPCD' - 범주형

In [ ]:
# 원-핫 인코딩으로 만들어 줌
dumm6 = pd.get_dummies(df['TPGRP_TPCD'])
dumm6

0   2   3   7   10  12
0     0   0   0   0   0   1
1     0   0   0   0   0   1
2     0   0   0   1   0   0
3     0   0   0   0   0   1
4     0   0   0   0   0   1
..   ..  ..  ..  ..  ..  ..
632   1   0   0   0   0   0
633   1   0   0   0   0   0
634   1   0   0   0   0   0
635   1   0   0   0   0   0
636   1   0   0   0   0   0

[637 rows x 6 columns]

In [ ]:
# 데이터를 붙여줌
check_data6 = dumm6.join(land, how='inner')
check_data6

0  2  3  7  10  12  landslide
0    0  0  0  0   0   1        0.0
1    0  0  0  0   0   1        0.0
2    0  0  0  1   0   0        1.0
3    0  0  0  0   0   1        0.0
4    0  0  0  0   0   1        0.0
..  .. .. .. ..  ..  ..        ...
632  1  0  0  0   0   0        0.0
633  1  0  0  0   0   0        0.0
634  1  0  0  0   0   0        0.0
635  1  0  0  0   0   0        0.0
636  1  0  0  0   0   0        0.0

[637 rows x 7 columns]

In [ ]:
check_data6.corr()

0         2         3  ...        10        12  landslide
0          1.000000 -0.044141 -0.034466  ... -0.021305 -0.246568  -0.073728
2         -0.044141  1.000000 -0.037242  ... -0.023021 -0.266434  -0.123184
3         -0.034466 -0.037242  1.000000  ... -0.017975 -0.208034  -0.132587
7         -0.125372 -0.135474 -0.105779  ... -0.065386 -0.756746  -0.113457
10        -0.021305 -0.023021 -0.017975  ...  1.000000 -0.128594  -0.081958
12        -0.246568 -0.266434 -0.208034  ... -0.128594  1.000000   0.247476
landslide -0.073728 -0.123184 -0.132587  ... -0.081958  0.247476   1.000000

[7 rows x 7 columns]

In [ ]:
check_data6_vc = df[['TPGRP_TPCD', 'landslide']]

In [ ]:
check_data6_vc[check_data6_vc['landslide']==1].value_counts()

TPGRP_TPCD  landslide
12          1.0          181
7           1.0           51
0           1.0            5
2           1.0            3
dtype: int64

In [ ]:
check_data6_vc[check_data6_vc['landslide']==0].value_counts()

TPGRP_TPCD  landslide
12          0.0          200
7           0.0          126
2           0.0           26
0           0.0           20
3           0.0           18
10          0.0            7
dtype: int64

12와 7이 일어날 때 많이 발생하는 경향을 보인다.

# 7. 'PRDN_FOM_C' - 범주형

In [ ]:
# 원-핫 인코딩으로 만들어 줌
dumm7 = pd.get_dummies(df['PRDN_FOM_C'])
dumm7

0  1  2  3  4  5  6
0    0  1  0  0  0  0  0
1    0  1  0  0  0  0  0
2    0  1  0  0  0  0  0
3    0  1  0  0  0  0  0
4    0  1  0  0  0  0  0
..  .. .. .. .. .. .. ..
632  1  0  0  0  0  0  0
633  1  0  0  0  0  0  0
634  1  0  0  0  0  0  0
635  1  0  0  0  0  0  0
636  1  0  0  0  0  0  0

[637 rows x 7 columns]

In [ ]:
# 데이터를 붙여줌
check_data7 = dumm7.join(land, how='inner')
check_data7

0  1  2  3  4  5  6  landslide
0    0  1  0  0  0  0  0        0.0
1    0  1  0  0  0  0  0        0.0
2    0  1  0  0  0  0  0        1.0
3    0  1  0  0  0  0  0        0.0
4    0  1  0  0  0  0  0        0.0
..  .. .. .. .. .. .. ..        ...
632  1  0  0  0  0  0  0        0.0
633  1  0  0  0  0  0  0        0.0
634  1  0  0  0  0  0  0        0.0
635  1  0  0  0  0  0  0        0.0
636  1  0  0  0  0  0  0        0.0

[637 rows x 8 columns]

In [ ]:
check_data7.corr()

0         1         2  ...         5         6  landslide
0          1.000000 -0.260818 -0.017977  ... -0.091831 -0.070444  -0.073728
1         -0.260818  1.000000 -0.114781  ... -0.586326 -0.449773  -0.053214
2         -0.017977 -0.114781  1.000000  ... -0.040413 -0.031001  -0.069157
3         -0.044141 -0.281832 -0.019426  ... -0.099230 -0.076120  -0.014395
4         -0.011343 -0.072422 -0.004992  ... -0.025499 -0.019560  -0.043635
5         -0.091831 -0.586326 -0.040413  ...  1.000000 -0.158360   0.042430
6         -0.070444 -0.449773 -0.031001  ... -0.158360  1.000000   0.114698
landslide -0.073728 -0.053214 -0.069157  ...  0.042430  0.114698   1.000000

[8 rows x 8 columns]

In [ ]:
check_data7_vc = df[['PRDN_FOM_C', 'landslide']]

In [ ]:
check_data7_vc[check_data7_vc['landslide']==1].value_counts()

PRDN_FOM_C  landslide
1           1.0          142
5           1.0           46
6           1.0           37
3           1.0           10
0           1.0            5
dtype: int64

In [ ]:
check_data7_vc[check_data7_vc['landslide']==0].value_counts()

PRDN_FOM_C  landslide
1           0.0          256
5           0.0           63
6           0.0           32
0           0.0           20
3           0.0           19
2           0.0            5
4           0.0            2
dtype: int64

규칙을 못찾겠다..

# 8. 'SLANT_TYP' - 범주형

In [ ]:
# 원-핫 인코딩으로 만들어 줌
dumm8 = pd.get_dummies(df['SLANT_TYP'])
dumm8

0  1  2  3
0    0  1  0  0
1    0  1  0  0
2    0  1  0  0
3    0  1  0  0
4    0  1  0  0
..  .. .. .. ..
632  1  0  0  0
633  1  0  0  0
634  1  0  0  0
635  1  0  0  0
636  1  0  0  0

[637 rows x 4 columns]

In [ ]:
# 데이터를 붙여줌
check_data8 = dumm8.join(land, how='inner')
check_data8

0  1  2  3  landslide
0    0  1  0  0        0.0
1    0  1  0  0        0.0
2    0  1  0  0        1.0
3    0  1  0  0        0.0
4    0  1  0  0        0.0
..  .. .. .. ..        ...
632  1  0  0  0        0.0
633  1  0  0  0        0.0
634  1  0  0  0        0.0
635  1  0  0  0        0.0
636  1  0  0  0        0.0

[637 rows x 5 columns]

In [ ]:
check_data8.corr()

0         1         2         3  landslide
0          1.000000 -0.259946 -0.100362 -0.081451  -0.073728
1         -0.259946  1.000000 -0.638652 -0.518316  -0.050651
2         -0.100362 -0.638652  1.000000 -0.200115  -0.003843
3         -0.081451 -0.518316 -0.200115  1.000000   0.116505
landslide -0.073728 -0.050651 -0.003843  0.116505   1.000000

In [ ]:
check_data8_vc = df[['SLANT_TYP', 'landslide']]

In [ ]:
check_data8_vc[check_data8_vc['landslide']==1].value_counts()

SLANT_TYP  landslide
1          1.0          142
2          1.0           47
3          1.0           46
0          1.0            5
dtype: int64

In [ ]:
check_data8_vc[check_data8_vc['landslide']==0].value_counts()

SLANT_TYP  landslide
1          0.0          255
2          0.0           79
3          0.0           43
0          0.0           20
dtype: int64

In [ ]:
# 각 index별로 나눗셈을 보는 과정

values = check_data8_vc[check_data8_vc['landslide']==1].value_counts().tolist()
keys = check_data8_vc[check_data8_vc['landslide']==1].value_counts().index.tolist()

indexs = [keys_index[0] for keys_index in keys]
print(indexs)

A1 = dict(zip(indexs, values)) # 1

values2 = check_data8_vc[check_data8_vc['landslide']==0].value_counts().tolist()
keys2 = check_data8_vc[check_data8_vc['landslide']==0].value_counts().index.tolist()

indexs2 = [keys_index[0] for keys_index in keys2]
print(indexs2)

A2 = dict(zip(indexs2, values2)) # 0

B1 = dict(sorted(A1.items())) # 1
B2 = dict(sorted(A2.items())) # 0
print(B1, B2)

from collections import Counter
C = dict(Counter(B1) + Counter(B2)) # 1과 0의 전체 합

print(C)

final_result = {x:float(B1[x])/C[x] for x in C}
print(final_result)

[1, 2, 3, 0]
[1, 2, 3, 0]
{0: 5, 1: 142, 2: 47, 3: 46} {0: 20, 1: 255, 2: 79, 3: 43}
{0: 25, 1: 397, 2: 126, 3: 89}
{0: 0.2, 1: 0.35768261964735515, 2: 0.373015873015873, 3: 0.5168539325842697}


규칙을 못찾겠다..

# 9. 'SLDPT_TPCD' - 범주형

In [ ]:
# 원-핫 인코딩으로 만들어 줌
dumm9 = pd.get_dummies(df['SLDPT_TPCD'])
dumm9

0   1   2   3   10  20  30
0     0   0   0   0   0   1   0
1     0   0   0   0   1   0   0
2     0   0   0   0   1   0   0
3     0   0   0   0   1   0   0
4     0   0   0   0   1   0   0
..   ..  ..  ..  ..  ..  ..  ..
632   1   0   0   0   0   0   0
633   1   0   0   0   0   0   0
634   1   0   0   0   0   0   0
635   1   0   0   0   0   0   0
636   1   0   0   0   0   0   0

[637 rows x 7 columns]

In [ ]:
# 데이터를 붙여줌
check_data9 = dumm9.join(land, how='inner')
check_data9

0  1  2  3  10  20  30  landslide
0    0  0  0  0   0   1   0        0.0
1    0  0  0  0   1   0   0        0.0
2    0  0  0  0   1   0   0        1.0
3    0  0  0  0   1   0   0        0.0
4    0  0  0  0   1   0   0        0.0
..  .. .. .. ..  ..  ..  ..        ...
632  1  0  0  0   0   0   0        0.0
633  1  0  0  0   0   0   0        0.0
634  1  0  0  0   0   0   0        0.0
635  1  0  0  0   0   0   0        0.0
636  1  0  0  0   0   0   0        0.0

[637 rows x 8 columns]

In [ ]:
check_data9.corr()

0         1         2  ...        20        30  landslide
0          1.000000 -0.034466 -0.041693  ... -0.109723 -0.074946  -0.073728
1         -0.034466  1.000000 -0.035177  ... -0.092575 -0.063233  -0.132587
2         -0.041693 -0.035177  1.000000  ... -0.111987 -0.076492  -0.127643
3         -0.013903 -0.011730 -0.014190  ... -0.037344 -0.025507  -0.006166
10        -0.218307 -0.184189 -0.222812  ... -0.586374 -0.400520  -0.007998
20        -0.109723 -0.092575 -0.111987  ...  1.000000 -0.201304   0.018303
30        -0.074946 -0.063233 -0.076492  ... -0.201304  1.000000   0.178775
landslide -0.073728 -0.132587 -0.127643  ...  0.018303  0.178775   1.000000

[8 rows x 8 columns]

In [ ]:
check_data9_vc = df[['SLDPT_TPCD', 'landslide']]

In [ ]:
check_data9_vc[check_data9_vc['landslide']==1].value_counts()

SLDPT_TPCD  landslide
10          1.0          128
20          1.0           57
30          1.0           47
0           1.0            5
2           1.0            2
3           1.0            1
dtype: int64

In [ ]:
check_data9_vc[check_data9_vc['landslide']==0].value_counts()

SLDPT_TPCD  landslide
10          0.0          215
20          0.0           88
30          0.0           30
2           0.0           24
0           0.0           20
1           0.0           18
3           0.0            2
dtype: int64

30이 많을 수록 1도 많이 발생

# 10. 'SCSTX_CD' - 범주형

In [ ]:
# 원-핫 인코딩으로 만들어 줌
dumm10 = pd.get_dummies(df['SCSTX_CD'])
dumm10

0  1  2  3  4  9
0    0  0  1  0  0  0
1    0  0  1  0  0  0
2    0  0  0  1  0  0
3    0  0  1  0  0  0
4    0  0  1  0  0  0
..  .. .. .. .. .. ..
632  1  0  0  0  0  0
633  1  0  0  0  0  0
634  1  0  0  0  0  0
635  1  0  0  0  0  0
636  1  0  0  0  0  0

[637 rows x 6 columns]

In [ ]:
# 데이터를 붙여줌
check_data10 = dumm10.join(land, how='inner')
check_data10

0  1  2  3  4  9  landslide
0    0  0  1  0  0  0        0.0
1    0  0  1  0  0  0        0.0
2    0  0  0  1  0  0        1.0
3    0  0  1  0  0  0        0.0
4    0  0  1  0  0  0        0.0
..  .. .. .. .. .. ..        ...
632  1  0  0  0  0  0        0.0
633  1  0  0  0  0  0        0.0
634  1  0  0  0  0  0        0.0
635  1  0  0  0  0  0        0.0
636  1  0  0  0  0  0        0.0

[637 rows x 7 columns]

In [ ]:
check_data10.corr()

0         1         2  ...         4         9  landslide
0          1.000000 -0.125863 -0.156098  ... -0.019709 -0.028006  -0.073728
1         -0.125863  1.000000 -0.480956  ... -0.060725 -0.086289  -0.029344
2         -0.156098 -0.480956  1.000000  ... -0.075312 -0.107017   0.075872
3         -0.125863 -0.387800 -0.480956  ... -0.060725 -0.086289  -0.051003
4         -0.019709 -0.060725 -0.075312  ...  1.000000 -0.013512   0.024799
9         -0.028006 -0.086289 -0.107017  ... -0.013512  1.000000   0.082897
landslide -0.073728 -0.029344  0.075872  ...  0.024799  0.082897   1.000000

[7 rows x 7 columns]

In [ ]:
check_data10_vc = df[['SCSTX_CD', 'landslide']]

In [ ]:
check_data10_vc[check_data10_vc['landslide']==1].value_counts()

SCSTX_CD  landslide
2         1.0          101
1         1.0           63
3         1.0           60
9         1.0            8
0         1.0            5
4         1.0            3
dtype: int64

In [ ]:
check_data10_vc[check_data10_vc['landslide']==0].value_counts()

SCSTX_CD  landslide
2         0.0          137
3         0.0          118
1         0.0          115
0         0.0           20
9         0.0            4
4         0.0            3
dtype: int64

In [ ]:
# 각 index별로 나눗셈을 보는 과정

values = check_data10_vc[check_data10_vc['landslide']==1].value_counts().tolist()
keys = check_data10_vc[check_data10_vc['landslide']==1].value_counts().index.tolist()

indexs = [keys_index[0] for keys_index in keys]
print(indexs)

A1 = dict(zip(indexs, values)) # 1

values2 = check_data10_vc[check_data10_vc['landslide']==0].value_counts().tolist()
keys2 = check_data10_vc[check_data10_vc['landslide']==0].value_counts().index.tolist()

indexs2 = [keys_index[0] for keys_index in keys2]
print(indexs2)

A2 = dict(zip(indexs2, values2)) # 0

B1 = dict(sorted(A1.items())) # 1
B2 = dict(sorted(A2.items())) # 0
print(B1, B2)

from collections import Counter
C = dict(Counter(B1) + Counter(B2)) # 1과 0의 전체 합

print(C)

final_result = {x:float(B1[x])/C[x] for x in C}
print(final_result)

[2, 1, 3, 9, 0, 4]
[2, 3, 1, 0, 9, 4]
{0: 5, 1: 63, 2: 101, 3: 60, 4: 3, 9: 8} {0: 20, 1: 115, 2: 137, 3: 118, 4: 3, 9: 4}
{0: 25, 1: 178, 2: 238, 3: 178, 4: 6, 9: 12}
{0: 0.2, 1: 0.3539325842696629, 2: 0.42436974789915966, 3: 0.33707865168539325, 4: 0.5, 9: 0.6666666666666666}


대다수가 생각보다 연관성이 적음

# 11. 'SLTP_CD' - 범주형

In [ ]:
# 원-핫 인코딩으로 만들어 줌
dumm11 = pd.get_dummies(df['SLTP_CD'])
dumm11

0   1   2   3   5   6   10  13  14  15  16  24  26  27
0     0   1   0   0   0   0   0   0   0   0   0   0   0   0
1     0   1   0   0   0   0   0   0   0   0   0   0   0   0
2     0   0   0   0   0   0   0   1   0   0   0   0   0   0
3     0   1   0   0   0   0   0   0   0   0   0   0   0   0
4     0   1   0   0   0   0   0   0   0   0   0   0   0   0
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..
632   1   0   0   0   0   0   0   0   0   0   0   0   0   0
633   1   0   0   0   0   0   0   0   0   0   0   0   0   0
634   1   0   0   0   0   0   0   0   0   0   0   0   0   0
635   1   0   0   0   0   0   0   0   0   0   0   0   0   0
636   1   0   0   0   0   0   0   0   0   0   0   0   0   0

[637 rows x 14 columns]

In [ ]:
# 데이터를 붙여줌
check_data11 = dumm11.join(land, how='inner')
check_data11

0  1  2  3  5  6  10  13  14  15  16  24  26  27  landslide
0    0  1  0  0  0  0   0   0   0   0   0   0   0   0        0.0
1    0  1  0  0  0  0   0   0   0   0   0   0   0   0        0.0
2    0  0  0  0  0  0   0   1   0   0   0   0   0   0        1.0
3    0  1  0  0  0  0   0   0   0   0   0   0   0   0        0.0
4    0  1  0  0  0  0   0   0   0   0   0   0   0   0        0.0
..  .. .. .. .. .. ..  ..  ..  ..  ..  ..  ..  ..  ..        ...
632  1  0  0  0  0  0   0   0   0   0   0   0   0   0        0.0
633  1  0  0  0  0  0   0   0   0   0   0   0   0   0        0.0
634  1  0  0  0  0  0   0   0   0   0   0   0   0   0        0.0
635  1  0  0  0  0  0   0   0   0   0   0   0   0   0        0.0
636  1  0  0  0  0  0   0   0   0   0   0   0   0   0        0.0

[637 rows x 15 columns]

In [ ]:
check_data11.corr()

0         1         2  ...        26        27  landslide
0          1.000000 -0.183602 -0.116079  ... -0.037318 -0.016067  -0.073728
1         -0.183602  1.000000 -0.521728  ... -0.167727 -0.072212   0.003198
2         -0.116079 -0.521728  1.000000  ... -0.106043 -0.045655   0.123555
3         -0.022794 -0.102448 -0.064771  ... -0.020823 -0.008965   0.057772
5         -0.017977 -0.080800 -0.051084  ... -0.016423 -0.007071  -0.032446
6         -0.008014 -0.036021 -0.022774  ... -0.007321 -0.003152   0.050999
10        -0.008014 -0.036021 -0.022774  ... -0.007321 -0.003152  -0.030831
13        -0.051615 -0.231988 -0.146670  ... -0.047152 -0.020301  -0.103968
14        -0.039118 -0.175818 -0.111158  ... -0.035735 -0.015385   0.023172
15        -0.050906 -0.228803 -0.144657  ... -0.046505 -0.020022  -0.004338
16        -0.037318 -0.167727 -0.106043  ... -0.034091 -0.014677  -0.070982
24        -0.017977 -0.080800 -0.051084  ... -0.016423 -0.007071  -0.032446
26        -0.037318 -0.167727 -0.106043  ...  1.000000 -0.014677  -0.016549
27        -0.016067 -0.072212 -0.045655  ... -0.014677  1.000000  -0.061807
landslide -0.073728  0.003198  0.123555  ... -0.016549 -0.061807   1.000000

[15 rows x 15 columns]

In [ ]:
check_data11_vc = df[['SLTP_CD', 'landslide']]

In [ ]:
check_data11_vc[check_data11_vc['landslide']==1].value_counts()

SLTP_CD  landslide
1        1.0          109
2        1.0           76
15       1.0           14
14       1.0           10
26       1.0            7
13       1.0            7
3        1.0            5
0        1.0            5
16       1.0            4
24       1.0            1
6        1.0            1
5        1.0            1
dtype: int64

In [ ]:
check_data11_vc[check_data11_vc['landslide']==0].value_counts()

SLTP_CD  landslide
1        0.0          179
2        0.0           82
13       0.0           32
15       0.0           24
0        0.0           20
16       0.0           17
26       0.0           14
14       0.0           13
27       0.0            4
24       0.0            4
5        0.0            4
3        0.0            3
10       0.0            1
dtype: int64

대다수가 생각보다 연관성이 적음, 2 정도가 연관성이 좀 있어 보임.

## 12. 'EIGHT_AGL_dir' - 범주형

In [ ]:
# 원-핫 인코딩으로 만들어 줌
dumm12 = pd.get_dummies(df['EIGHT_AGL_dir'])
dumm12

E  N  S  W
0    0  0  1  0
1    0  0  0  1
2    0  0  0  1
3    0  1  0  0
4    1  0  0  0
..  .. .. .. ..
632  0  1  0  0
633  0  1  0  0
634  0  1  0  0
635  0  1  0  0
636  0  1  0  0

[637 rows x 4 columns]

In [ ]:
# 데이터를 붙여줌
check_data12 = dumm12.join(land, how='inner')
check_data12

E  N  S  W  landslide
0    0  0  1  0        0.0
1    0  0  0  1        0.0
2    0  0  0  1        1.0
3    0  1  0  0        0.0
4    1  0  0  0        0.0
..  .. .. .. ..        ...
632  0  1  0  0        0.0
633  0  1  0  0        0.0
634  0  1  0  0        0.0
635  0  1  0  0        0.0
636  0  1  0  0        0.0

[637 rows x 5 columns]

In [ ]:
check_data12.corr()

E         N         S         W  landslide
E          1.000000 -0.353766 -0.309953 -0.343876   0.044367
N         -0.353766  1.000000 -0.321507 -0.356695  -0.032364
S         -0.309953 -0.321507  1.000000 -0.312520  -0.013669
W         -0.343876 -0.356695 -0.312520  1.000000   0.001559
landslide  0.044367 -0.032364 -0.013669  0.001559   1.000000

In [ ]:
check_data12_vc = df[['EIGHT_AGL_dir', 'landslide']]

In [ ]:
check_data12_vc[check_data12_vc['landslide']==1].value_counts()

EIGHT_AGL_dir  landslide
E              1.0          67
W              1.0          62
N              1.0          60
S              1.0          51
dtype: int64

In [ ]:
check_data12_vc[check_data12_vc['landslide']==0].value_counts()

EIGHT_AGL_dir  landslide
N              0.0          111
W              0.0          102
E              0.0           95
S              0.0           89
dtype: int64

너무 골고루임.

In [ ]:
# 각 index별로 나눗셈을 보는 과정

values = check_data12_vc[check_data12_vc['landslide']==1].value_counts().tolist()
keys = check_data12_vc[check_data12_vc['landslide']==1].value_counts().index.tolist()

indexs = [keys_index[0] for keys_index in keys]
print(indexs)

A1 = dict(zip(indexs, values)) # 1

values2 = check_data12_vc[check_data12_vc['landslide']==0].value_counts().tolist()
keys2 = check_data12_vc[check_data12_vc['landslide']==0].value_counts().index.tolist()

indexs2 = [keys_index[0] for keys_index in keys2]
print(indexs2)

A2 = dict(zip(indexs2, values2)) # 0

B1 = dict(sorted(A1.items())) # 1
B2 = dict(sorted(A2.items())) # 0
print(B1, B2)

from collections import Counter
C = dict(Counter(B1) + Counter(B2)) # 1과 0의 전체 합

print(C)

final_result = {x:float(B1[x])/C[x] for x in C}
print(final_result)

['E', 'W', 'N', 'S']
['N', 'W', 'E', 'S']
{'E': 67, 'N': 60, 'S': 51, 'W': 62} {'E': 95, 'N': 111, 'S': 89, 'W': 102}
{'E': 162, 'N': 171, 'S': 140, 'W': 164}
{'E': 0.41358024691358025, 'N': 0.3508771929824561, 'S': 0.36428571428571427, 'W': 0.3780487804878049}
